src/lib/comm.ts

```diff
@@ -26,9 +26,19 @@ export function getJson(options: CommOptions): Promise <any> {
       body = await res.json();
       // TODO: better error messaging here
       if (!res.ok) {
-        reject(new Error(body && body.message ? body.message
-          : 'Server sent error code ' + res.status + '\n\n' + JSON.stringify(body)));
+        if (body && body.message) {
+          let message = body.message;
+
+          if (Object.prototype.toString.call(message).indexOf('Object') > -1 ) {
```

Comment:
can we change this to ```if (typeof message === 'object')``` ? correct me if i'm wrong but i think that would be cleaner.

also, lets put the following comment aboev this line
// Current blueprint/idm API does not have standardized error responses, some messages, such as bad password error from sign up, return an object and not a string. 
// TODO: Document all error formats and account for them in a clean fashion.